satpy/tests/reader_tests/test_landsat.py

```diff
@@ -0,0 +1,3784 @@
+    return os.fspath(filename)
+
+@pytest.fixture(scope="session")
+def tm_l2_rad_file(tm_l2_files_path, sza_rad_data, tm_l2_area):
+    """Create the file for the Landsat TM L2 TRAD channel."""
+    filename = tm_l2_files_path / "LT05_L2SP_165054_20110817_20200820_02_T1_ST_TRAD.TIF"
+    create_tif_file(sza_rad_data, "TRAD", tm_l2_area, filename, tm_l2_date)
+    return os.fspath(filename)
+
+
+@pytest.fixture(scope="session")
+def tm_l2_mda_file(tm_l2_files_path):
+    """Create the Landsat TM L2 metadata xml file."""
+    filename = tm_l2_files_path / "LT05_L2SP_165054_20110817_20200820_02_T1_MTL.xml"
+    with open(filename, "wb") as f:
+        f.write(tm_l2_metadata_text)
+    return os.fspath(filename)
+
+
+@pytest.fixture(scope="session")
+def tm_l2_all_files(tm_l2_b4_file, tm_l2_b6_file, tm_l2_mda_file, tm_l2_rad_file):
+    """Return all the files."""
+    return tm_l2_b4_file, tm_l2_b6_file, tm_l2_mda_file, tm_l2_rad_file
+
+
+@pytest.fixture(scope="session")
+def mss_l1_landsat1_b4_file(mss_l1_landsat1_files_path, spectral_data, mss_l1_landsat1_area):
+    """Create the file for the Landsat-1 MSS L1 B4 channel."""
+    filename = mss_l1_landsat1_files_path / "LM01_L1TP_032030_19720729_20200909_02_T2_B4.TIF"
+    create_tif_file(spectral_data, "B4", mss_l1_landsat1_area, filename, mss_l1_landsat1_date)
+    return os.fspath(filename)
+
+
+@pytest.fixture(scope="session")
+def mss_l1_landsat1_mda_file(mss_l1_landsat1_files_path):
+    """Create the Landsat-1 MSS L1 metadata xml file."""
+    filename = mss_l1_landsat1_files_path / "LM01_L1TP_032030_19720729_20200909_02_T2_MTL.xml"
+    with open(filename, "wb") as f:
+        f.write(mss_l1_landsat1_metadata_text)
+    return os.fspath(filename)
+
+
+@pytest.fixture(scope="session")
+def mss_l1_landsat1_all_files(mss_l1_landsat1_b4_file, mss_l1_landsat1_mda_file):
+    """Return all the files."""
+    return mss_l1_landsat1_b4_file, mss_l1_landsat1_mda_file
+
+
+@pytest.fixture(scope="session")
+def mss_l1_landsat4_b4_file(mss_l1_landsat4_files_path, spectral_data, mss_l1_landsat4_area):
+    """Create the file for the Landsat-4 MSS L1 B4 channel."""
+    filename = mss_l1_landsat4_files_path / "LM04_L1TP_029030_19840415_20200903_02_T2_B4.TIF"
+    create_tif_file(spectral_data, "B4", mss_l1_landsat4_area, filename, mss_l1_landsat4_date)
+    return os.fspath(filename)
+
+
+@pytest.fixture(scope="session")
+def mss_l1_landsat4_mda_file(mss_l1_landsat4_files_path):
+    """Create the Landsat-4 MSS L1 metadata xml file."""
+    filename = mss_l1_landsat4_files_path / "LM04_L1TP_029030_19840415_20200903_02_T2_MTL.xml"
+    with open(filename, "wb") as f:
+        f.write(mss_l1_landsat4_metadata_text)
+    return os.fspath(filename)
+
+
+@pytest.fixture(scope="session")
+def mss_l1_landsat4_all_files(mss_l1_landsat4_b4_file, mss_l1_landsat4_mda_file):
+    """Return all the files."""
+    return mss_l1_landsat4_b4_file, mss_l1_landsat4_mda_file
+
+
+@pytest.fixture(scope="session")
```

Comment:
Are these fixtures used in the other test modules? The scope for all of these should be "module". If these are used in the other landsat test modules then how about:

1. Create a `satpy/tests/reader_tests/landsat_tests/` directory.
2. Move these fixtures to `landsat_tests/conftest.py` module and set scope to "module".
3. Move all the landsat tests to `landsat_tests/` directory.
4. Use the fixtures as usual. Pytest will automatically load them due to their proximity to `conftest.py`.

And yes, they will be recreated for each test module, but in my opinion that's better than them existing for all tests for the whole session.